### PR TITLE
implemented quick union weighted

### PR DESCRIPTION
**Pull request by tatsiak**
Homework which I haven't done in time.

### CODE REVIEW

File: app/quickUnionWeighted.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        if (isNaN(this._elements[a])) this._elements[a] = a;
+        if (isNaN(this._elements[b])) this._elements[b] = b;
+
+        let parentA = this.component(a, this._elements);
+        let parentB = this.component(b, this._elements);
+
+        if (isNaN(this._size[parentA])) this._size[parentA] = 1;
+        if (isNaN(this._size[parentB])) this._size[parentB] = 1;
+
+        if (this._size[parentA] > this._size[parentB]) {
```
Comment (pavlom0x) [inline app/quickUnionWeighted.js]: You should connect smaller component to bigger
and you should have a test that checked it

File: test/quickUnionWeighted.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            (uf.component(91) === undefined).should.be.true();
+        })
+
+        it('should return component name', function () {
+            let uf = new algolib.UnionFind();
+            uf.connect(1, 3);
+            uf.connect(3, 6);
+            uf.connect(6, 9);
+            let componentName = uf.component(1);
+            componentName.should.be.eql(9);
```
Comment (pavlom0x) [inline test/quickUnionWeighted.spec.js]: it should not return `9`, it should return `1` or `3` depend on your implementation
method `uf.component` implement wrong

it should go from element to it parent until index and value equal each other
try to draw how it work on a piece of paper

**Review by pavlom0x — CHANGES_REQUESTED** [review]

File: app/quickUnionWeighted.js (hunk 1 above)
Comment (tatsiak) [reply app/quickUnionWeighted.js]: fixed

File: test/quickUnionWeighted.spec.js (hunk 2 above)
Comment (tatsiak) [reply test/quickUnionWeighted.spec.js]: fixed

**Review by pavlom0x — APPROVED** [review]

**Comment by pavlom0x** [comment]
you can merge it

**Pull request merged**